### PR TITLE
Add a circuit breaker consumer to GeoGrid aggregations

### PR DESCRIPTION
**Pull request by iverase**
We are currently only adding a consumer for grid aggregations on geo_shape fields. In the case of multivalued geo_points, we are actually creating an array but it is not backed by a circuit breaker. This PR fixes that.

In addition, they way we are setting the consumer for geo_shape is quite hacky. This PR proposes to change the ValueSource parameter in the constructor with a `Function<LongConsumer, ValuesSource.Numeric>` , therefore we can set the consumer inside the constructor which makes code simpler.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/CellIdSource.java [hunk 1]
```diff
@@ -119,7 +123,8 @@ protected abstract static class CellMultiValues extends AbstractSortingNumericDo
         private final MultiGeoPointValues geoValues;
         protected final int precision;
 
-        protected CellMultiValues(MultiGeoPointValues geoValues, int precision) {
+        protected CellMultiValues(MultiGeoPointValues geoValues, int precision, LongConsumer circuitBreakerConsumer) {
+            super(circuitBreakerConsumer);
```
Comment (iverase) [inline server/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/CellIdSource.java:127]: This looks better.

**Comment by iverase** [comment]
@elasticmachine run elasticsearch-ci/part-1

**Review by not-napoleon — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/GeoTileGridValuesSourceBuilder.java [hunk 2]
```diff
@@ -82,7 +82,8 @@ static void register(ValuesSourceRegistry.Builder builder) {
                 ValuesSource.GeoPoint geoPoint = (ValuesSource.GeoPoint) valuesSourceConfig.getValuesSource();
                 // is specified in the builder.
                 final MappedFieldType fieldType = valuesSourceConfig.fieldType();
-                GeoTileCellIdSource cellIdSource = new GeoTileCellIdSource(geoPoint, precision, boundingBox);
+                // TODO: support adding circuit breaker in composite
```
[the remark this answers is not in the document]
Comment (iverase) [reply server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/GeoTileGridValuesSourceBuilder.java:85]: I open https://github.com/elastic/elasticsearch/issues/92861

**Pull request merged**